File: benchmarks/reference/rust/udiv_stress.rs
fn main() {
    let mut sum: i64 = 0;
    for i in 1..500001_i64 {
        let d1 = i / 2;
        let d2 = i / 4;
        let d3 = i / 8;
        let d4 = i / 16;
        let d5 = i / 32;
        let d6 = i / 64;
        let d7 = i / 128;
        let d8 = i / 256;
        let s7 = d1 + d2 + d3 + d4 + d5 + d6 + d7 + d8;
        let raw_sum = sum + s7;
        sum = raw_sum & 268435455;
    }
    std::process::exit((sum & 0xFF) as i32);
}

File: benchmarks/reference/csharp/UdivStress.cs
/* udiv_stress — Unsigned division stress benchmark (50M iterations).
   Equivalent to examples/il/benchmarks/udiv_stress.il */
using System;
using System.Runtime.CompilerServices;

class UdivStress
{
    [MethodImpl(MethodImplOptions.NoInlining)]
    static long Run(long n)
    {
        long sum = 0;
        for (long i = 1; i < n; ++i)
        {
            long d1 = i / 2;
            long d2 = i / 4;
            long d3 = i / 8;
            long d4 = i / 16;
            long d5 = i / 32;
            long d6 = i / 64;
            long d7 = i / 128;
            long d8 = i / 256;

            long s7 = d1 + d2 + d3 + d4 + d5 + d6 + d7 + d8;

            long rawSum = sum + s7;
            sum = rawSum & 268435455;
        }
        return sum;
    }

    static int Main(string[] args)
    {
        long result = Run(50000001 + args.Length);
        return (int)(result & 0xFF);
    }
}
